listener.c

```diff
@@ -393,7 +393,7 @@ listener_read_cb(evutil_socket_t fd, short what, void *p)
 	evconnlistener_errorcb errorcb;
 	void *user_data;
 	LOCK(lev);
-	while (1) {
+	while (lev->enabled) {
```

Comment:
the intention was to immediately act, if a cb decides to disable the listener, to prevent it draining the queue.

If you would accept this as a separate PR I would add it

Reply:
yep, just submit another PR, since this is totally different thing